File: fink-admin/src/main/java/org/noorg/fink/admin/support/ApplicationContextProvider.java
package org.noorg.fink.admin.support;

import org.springframework.beans.BeansException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;

public class ApplicationContextProvider implements ApplicationContextAware {

	private static ApplicationContext context;

	public static ApplicationContext getContext() {
		return context;
	}
	
	@Override
	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
		context = applicationContext;
	}

}

File: fink-admin/src/main/scala/org/noorg/fink/admin/support/ApplicationContextProvider.scala
package org.noorg.fink.admin.support;

import org.springframework.beans.BeansException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;

class ApplicationContextProvider extends ApplicationContextAware {
	override def setApplicationContext(applicationContext: ApplicationContext) = {
		ApplicationContextProvider.context = applicationContext;
	}
}

object ApplicationContextProvider {

	private var context: ApplicationContext = null

	def getContext() = context

}
